finetune/nn/target_blocks.py

```diff
@@ -64,19 +63,43 @@ def masked_language_model(*, X, M, mlm_weights, mlm_positions, mlm_ids, embed_we
             transpose_b=True
         )
 
-        # final_proj_w = tf.get_variable(
-        #     'dense/kernel',
-        #     [config.n_embed, config.n_embed],
-        #     initializer=tf.random_normal_initializer(stddev=config.weight_stddev)
-        # )
-        # final_proj_b = tf.get_variable(
-        #     'dense/bias',
-        #     [config.n_embed],
-        #     initializer=tf.zeros_initializer
-        # )
-        # final_proj = act_fns[config.act_fn](
-        #     tf.matmul(gathered_hidden, final_proj_w, transpose_b=True) + final_proj_b
-        # )
+def masked_language_model(*, X, M, mlm_weights, mlm_ids, mlm_positions, embed_weights, hidden, config, reuse=None, train=False):
+    
+    with tf.variable_scope('model/masked-language-model'):
+        batch, seq, feats = shape_list(hidden)
+        flat_offsets = tf.reshape(
+            tf.range(0, batch, dtype=tf.int32) * seq, [-1, 1]
+        )
+
+        not_padding = tf.reshape(mlm_weights, [-1]) > 1e-9
+        flat_positions = tf.boolean_mask(tf.reshape(mlm_positions + flat_offsets, [-1]), not_padding) # take off the padding entirely
```

Comment:
why do you need the boolean mask? maybe I'm missing something, but doesn't `tf.reshape(mlm_positions + flat_offsets, [-1])` just give the positions, offset by the example in the batch?

Reply:
also, it looks like mlm_weights is just a tensor of 1s now, so the mask probably doesn't do anything?

Reply:
So, any padding added by padded_batch. this removes it.